deps/oblib/src/lib/charset/ob_ctype_utf8_os.cc

```diff
@@ -1696,29 +1696,40 @@ static int ob_wc_mb_utf8mb4(const ObCharsetInfo *cs __attribute__((unused)),
   if (OB_UNLIKELY(len <= 0)) {
     ret = OB_CS_TOOSMALL;
   } else if (w_char < 0x80) { //7	U+0000	U+007F	1	0xxxxxxx
-    bytes = 1;
+    if(OB_LIKELY(len >=1)){
+      ret = 1;
+      r[0] = (unsigned char) w_char;
+    }
+    else{
+      ret = OB_CS_TOOSMALLN(1);
```

Comment:
line 1702 and line 1703 should be together,  like "} else {", minded that there is empty space around 'else'